scripts/merge_users.py

```diff
@@ -0,0 +1,285 @@
+@click.option('-c', '--config', type=click.Path(exists=True),
+              help="Specify the path to a config file with database info.")
+@click.option('-L', '--live', is_flag=True, help='live run - use with extreme caution')
+@click.version_option(__version__)
+@click.pass_context
+def merge_users(context, mongodb, username, password, authdb, host, port, loglevel, config, live):
+    """scout: manage interactions with a scout instance."""
+    coloredlogs.install(level=loglevel)
+
+    LOG.info("Running scout version %s", __version__)
+    LOG.debug("Debug logging enabled.")
+
+    mongo_config = {}
+    cli_config = {}
+    if config:
+        LOG.debug("Use config file %s", config)
+        with open(config, 'r') as in_handle:
+            cli_config = yaml.load(in_handle, Loader=yaml.FullLoader)
+
+    mongo_config['mongodb'] = (mongodb or cli_config.get('mongodb') or 'scout')
+
+    mongo_config['host'] = (host or cli_config.get('host') or 'localhost')
+    mongo_config['port'] = (port or cli_config.get('port') or 27017)
+    mongo_config['username'] = username or cli_config.get('username')
+    mongo_config['password'] = password or cli_config.get('password')
+    mongo_config['authdb'] = authdb or cli_config.get('authdb') or mongo_config['mongodb']
+    mongo_config['omim_api_key'] = cli_config.get('omim_api_key')
+
+
+    # always dryrun for now
+    dryrun = True
+    if live:
+        dryrun = False
+
+    LOG.info("Setting database name to %s", mongo_config['mongodb'])
+    LOG.debug("Setting host to %s", mongo_config['host'])
+    LOG.debug("Setting port to %s", mongo_config['port'])
+
+    valid_connection = check_connection(
+        host=mongo_config['host'],
+        port=mongo_config['port'],
+        username=mongo_config['username'],
+        password=mongo_config['password'],
+        authdb=mongo_config['authdb'],
+    )
+
+    LOG.info("Test if mongod is running")
+    if not valid_connection:
+        LOG.warning("Connection could not be established")
+        context.abort()
+
+    try:
+        client = get_connection(**mongo_config)
+    except ConnectionFailure:
+        context.abort()
+
+    database = client[mongo_config['mongodb']]
+
+    LOG.info("Setting up a mongo adapter")
+    mongo_config['client'] = client
+    adapter = MongoAdapter(database)
+
+    user_requests = []
+    event_requests = []
+    acmg_requests = []
+    clinvar_requests = []
+    clinvar_submission_requests = []
+
+    for oi_user_obj in adapter.user_collection.find({'_id': {'$type':'objectId'}}):
+        if not oi_user_obj.get('email'):
+            continue
+
+        LOG.info('===USER===')
+        oi_user_id = oi_user_obj.get('_id')
+        oi_user_email = oi_user_obj.get('email')
+        LOG.info("user: {}".format(oi_user_obj))
+
+        create_alt = False
+        alt_user_obj = adapter.user_collection.find_one({'_id': oi_user_email})
+        if not alt_user_obj:
```

Comment:
I will agree that it is what caused the current spur of logins to the old accounts, but they have been around for a long time, and they have been used way-back-when as well.

Also, since we have had several users use their ObjectId over a few days now (e.g. Bianca, Sofie, Per, me..) there will be quite a few recent edits that would crash stuff/lack user names and other awkwardness if we just drop the users. 

So we do need to **at least** connect those events etc to their corresponding _id == email users.

Reply:
Perhaps if I tell you I could not dry run on the prod db without import copy?

Reply:
> So we do need to **at least** connect those events etc to their corresponding _id == email users.

I agree

Reply:
> Perhaps if I tell you I could not dry run on the prod db without import copy, and that the following block is the only place I use it?

I don't understand, what does it entail?

Reply:
Simply that it is being used when fed with live data, but only as edge cases.